backend/src/main/java/io/metersphere/api/service/ApiModuleService.java

```diff
@@ -632,174 +618,314 @@ public UpdateApiModuleDTO checkApiModule(ApiTestImportRequest request, ApiDefini
+                moduleMap = cover(moduleMap, toUpdateList, optionMap, repeatDataMap, updateVersionId, optionDataCases, oldCaseMap, esbApiParamsMap);
+            }
+        } else {
+            //不覆盖
+            removeRepeat(optionDatas, optionMap, repeatDataMap, moduleMap, versionId, optionDataCases);
+        }
 
-                //按照原来的顺序
-                methodPathMap = optionData.stream().collect(Collectors.toMap(t -> t.getMethod() + t.getPath(), api -> api));
+        //系统内检查重复
+        if (!repeatApiDefinitionWithBLOBs.isEmpty()) {
+            repeatDataMap = repeatApiDefinitionWithBLOBs.stream().collect(Collectors.toMap(t -> t.getName() + t.getModulePath(), api -> api));
+            optionMap = optionDatas.stream().collect(Collectors.toMap(t -> t.getName() + t.getModulePath(), api -> api));
+            if (fullCoverage) {
+                cover(moduleMap, toUpdateList, optionMap, repeatDataMap, updateVersionId, optionDataCases, oldCaseMap, esbApiParamsMap);
+            } else {
+                //不覆盖,同一接口不做更新
+                removeRepeat(optionDatas, optionMap, repeatDataMap, moduleMap, versionId, optionDataCases);
+            }
+        }
 
-                if (fullCoverage) {
-                    if (fullCoverageApi) {
-                        if (!repeatApiDefinitionWithBLOBs.isEmpty()) {
-                            startCoverModule(toUpdateList, optionData, methodPathMap, repeatDataMap, updateVersionId, optionDataCases, oldCaseMap);
-                        }
-                    } else {
-                        //不覆盖模块
-                        if (!repeatApiDefinitionWithBLOBs.isEmpty()) {
-                            if (repeatDataMap.size() >= methodPathMap.size()) {
-                                //导入文件没有新增接口无需创建接口模块
-                                moduleMap = new HashMap<>();
-                            }
-                            startCover(toUpdateList, optionData, methodPathMap, repeatDataMap, updateVersionId, optionDataCases, oldCaseMap);
-                        }
-                    }
-                } else {
-                    //不覆盖,同一接口不做更新
-                    if (!repeatApiDefinitionWithBLOBs.isEmpty()) {
-                        removeSameData(repeatDataMap, methodPathMap, optionData, moduleMap, versionId, optionDataCases);
-                    }
-                }
+        if (optionDatas.isEmpty()) {
+            moduleMap = new HashMap<>();
+        }
+        return getUpdateApiModuleDTO(moduleMap, toUpdateList, optionDatas, optionDataCases);
+    }
+
+    private void getNoHChooseModuleUrlRepeatOptionMap(List<ApiDefinitionWithBLOBs> optionDatas, Map<String, ApiDefinitionWithBLOBs> optionMap, String chooseModulePath) {
+        for (ApiDefinitionWithBLOBs optionDatum : optionDatas) {
+            if (optionDatum.getModulePath() == null) {
+                optionMap.put(optionDatum.getName() + chooseModulePath, optionDatum);
```

Comment:
StringUtils.concat

Reply:
Your suggestion is very helpful and accepted